tls/s2n_ktls.h

```diff
@@ -36,7 +37,16 @@ typedef enum {
 } s2n_ktls_mode;
 
 bool s2n_ktls_is_supported_on_platform();
+S2N_RESULT s2n_ktls_get_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
+S2N_RESULT s2n_ktls_sendmsg(struct s2n_connection *conn, struct msghdr *msg, ssize_t *send_len);
+S2N_RESULT s2n_ktls_recvmsg(struct s2n_connection *conn, struct msghdr *msg, ssize_t *bytes_read);
```

Comment:
These fn signatures will change once we implement ktls_send/recv. But for this PR I am keeping it simple so that we can test the IO mocking.